euphonic/spectra.py

```diff
@@ -555,11 +560,11 @@ def __add__(self, other: SC):
         spectra are retained in the top level dictionary, any
         others are put in the individual 'line_data' entries
         """
-        return Spectrum1DCollection.from_spectra([*self, *other])
+        return type(self).from_spectra([*self, *other])
```

Comment:
Function signature says it uses type of `other`, implementation says it uses type of `self`

Reply:
Is this alleviated if we do `__add__(self: T, other: T) -> T`, now they're all explicitly the same type?

Reply:
Perhaps it is still best to annotate it as `__add__(self: T, other: T) -> T` if we want to hide the fact that any iterable will work. It is an implementation detail that might get optimised. The inheritance scenario would still work, come to think of it.

Reply:
Mypy doesn't seem to complain either way, we can go with `Iterable`. Typing is hard.